File: backend/app/schemas/recommendation.py
from pydantic import BaseModel, Field

from app.schemas.place import Recommendation
from app.schemas.user import UserProfile


class RecommendationRequest(BaseModel):
    user_profile: UserProfile
    travel_date: str | None = Field(default=None, examples=["2026-07-10"])
    # 자연어 질의 (예: "전동휠체어인데 바다 보이는 산책로"). 없으면 기존 랭킹 그대로.
    query: str | None = Field(default=None, examples=["휠체어로 갈만한 해안 산책로"])


class RecommendationResponse(BaseModel):
    recommendations: list[Recommendation]
    # 질의가 실제 랭킹에 반영됐는지 (벡터DB 미준비·매칭 0건이면 False)
    query_applied: bool = False

File: frontend/src/api/recommendation.ts
import { client } from "./client";
import type { UserProfile } from "../types/user";
import type { Recommendation } from "../types/place";

export interface RecommendationRequest {
  user_profile: UserProfile;
  travel_date?: string;
  /** 자연어 질의 — 있으면 무장애 문서 관련도가 랭킹에 결합된다 */
  query?: string;
}

export interface RecommendationResponse {
  recommendations: Recommendation[];
  query_applied: boolean;
}

export async function postRecommendationsFull(
  payload: RecommendationRequest
): Promise<RecommendationResponse> {
  const { data } = await client.post<RecommendationResponse>(
    "/api/recommendations",
    payload
  );
  return data;
}

export async function postRecommendations(
  payload: RecommendationRequest
): Promise<Recommendation[]> {
  return (await postRecommendationsFull(payload)).recommendations;
}
